tests/UseCanonicalDataProvidersTest.hack

```diff
@@ -0,0 +1,42 @@
+/*
+ *  Copyright (c) 2017-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+
+namespace Facebook\HHAST;
+
+final class UseCanonicalDataProvidersTest extends TestCase {
+  use AutoFixingLinterTestTrait<ASTLintError>;
+
+  protected function getLinter(string $file): UseCanonicalDataProvidersLinter {
+    return UseCanonicalDataProvidersLinter::fromPath($file);
+  }
+
+  public function getCleanExamples(): vec<(string)> {
+    return vec[
+      tuple(<<<'REQUIRE_ONCE_FILE'
+use type Facebook\HackTest\{DataProvider, HackTest};
+use function Facebook\FBExpect\expect;
+
+final class ATestClass extends HackTest {
+  public function provideAdditions(): vec<(int, int, int)> {
```

Comment:
I'd like to see a test of `vec<(varray<int>)>` + darray

Reply:
Also `dict<...., (a, b, c)>`

Reply:
I'll add test cases for those.

Reply:
Fixed